docs/src/Experimental/plane_curves.md

```diff
@@ -0,0 +1,129 @@
+```@meta
+CurrentModule = Oscar
+```
+
+```@setup oscar
+using Oscar
+```
+
+```@contents
+Pages = ["plane_curves.md"]
+```
+
+
+# Affine and Projective Plane Curves
+
+We consider two kinds of plane curves : affine and projective. An affine plane curve is defined by a polynomial in two variables, whereas a projective plane curve is defined by an homogeneous polynomial belonging to a graded polynomial ring in three variables. 
+
+## Affine Plane Curves
+
+An affine plane curve is defined as the class of a two-variables polynomial ``F`` over a field ``K``, modulo the equivalence relation ``F \sim G \iff \exists \lambda \in K\backslash \{0\}, F = \lambda \cdot G``.
+
+### Example
+
+```@docs
+AffinePlaneCurve
+```
+
+## Projective Plane Curves
+
+Similarly, a projective plane curve is defined as the class of a three-variables homogeneous polynomial ``F`` over a field ``K``, modulo the equivalence relation ``F\sim G \iff \exists \lambda \in K\backslash \{0\}, F = \lambda \cdot G``. The defining equation is supposed to belong to a graded ring.
+
+```@docs
+ProjPlaneCurve
+```
+
+A particular kind of projective curves is the case of elliptic curves, see the corresponding section for more information. The types `ProjPlaneCurve` and `ProjEllipticCurve` are subtypes of the abstract type `ProjectivePlaneCurve`. In addition, the types `AffinePlaneCurve` and `ProjectivePlaneCurve` are subtypes of the abstract type `PlaneCurve`.
+
+## Points
+
+When considering curves, it is natural to have a look at points on the curve. We describe in this section how to deal with points, both in the affine and projective settings.
+
+### Point in the affine space
+
+A point in the affine space can be defined as follows:
+```@docs
+Point
+```
+We consider also the following function for points.
+
+```@docs
+ideal_point(R::MPolyRing{S}, P::Point{S}) where S <: FieldElem
+```
+
+The following function checks if a given point is on a curve:
+
+```@docs
+in(P::Point{S}, C::AffinePlaneCurve{S}) where S <: FieldElem
+```
+
+### Point in the projective space
+
+In order to define a point in the projective plane, one needs first to define the projective plane as follows, where `K` is the base ring:
+
+#### Example
+```@repl oscar
```

Comment:
Where possible, please consider using `jldoctest` instead of `@repl`. This has two advantages:
- we get some tests for free, and in particular catch if e.g. syntax changes (with `@repl` it can happen that one ends up with documentation saying things like "Error, no function projective_space" (say if that function got renamed, or was accidentally not exported, or...)
- since also output is included in the .md file, one can directly use the .md file to see an example, one does not have to render it